src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -640,10 +640,12 @@ private static void TrySchedule(PipeScheduler scheduler, Action<object> action,
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static void TrySchedule(PipeScheduler scheduler, in CompletionData completionData)
         {
+            Action<object> completion = completionData.Completion;
             // Nothing to do
-            if (completionData.Completion == null)
+            if (completion is null)
```

Comment:
why is?

Reply:
Smaller call path
```
(completion is null)
 => ReferenceEquals(completion, null)
```
vs
```
(completion == null)
 => MulticastDelegate.operator==(MulticastDelegate,MulticastDelegate)
 ==> ReferenceEquals(MulticastDelegate, MulticastDelegate)
 ==> ReferenceEquals(MulticastDelegate, null)
 ==> MulticastDelegate:Equals(ref):bool:this
```

Reply:
`MulticastDelegate.operator==` does inline now for the null and reference equals check https://github.com/dotnet/coreclr/pull/16431 but its still chunkier than just doing a `is` null or `ReferenceEquals` check (which don't call the virtual overload Equals methods)